universal-login-react/src/ui/Notifications/EmojiForm.tsx

```diff
@@ -32,76 +26,33 @@ interface EmojiFormProps {
 
 export const EmojiForm = ({deployedWallet, hideTitle, className, onDenyRequests, onConnectionSuccess}: EmojiFormProps) => {
   const [enteredCode, setEnteredCode] = useState<number[]>([]);
-  const [notifications, setNotifications] = useState<Notification[]>([]);
-  const [inputMode, setInputMode] = useState<InputModeType>('none');
-  const [addresses, setAddresses] = useState<string[]>([]);
   const {progressBar, showProgressBar} = useProgressBar();
-  const [isInputValid, setIsInputValid] = useState(false);
-  let gasParameters = INITIAL_GAS_PARAMETERS;
+  const [gasParameters, setGasParameters] = useState(INITIAL_GAS_PARAMETERS);
 
-  useEffect(() => deployedWallet.subscribeAuthorisations((notifications: Notification[]) => {
-    setNotifications(notifications);
-    updateAddressesAndInputMode(notifications);
-  }), []);
+  const [notifications, setNotifications] = useState<Notification[]>([]);
+  useEffect(() => deployedWallet.subscribeAuthorisations(setNotifications), []);
 
-  const updateAddressesAndInputMode = (notifications: Notification[]) => {
-    const addresses = filterNotificationByCodePrefix(notifications, enteredCode);
-    setInputMode(getInputModeFor(addresses, inputMode));
-    setAddresses(addresses);
-  };
+  const addresses = filterNotificationByCodePrefix(notifications, enteredCode);
+  const hint = addresses.length === 1 ? addresses[0] : undefined;
```

Comment:
Maybe change name `hint` to `isKeyboardNeeded` or `isMoreThanOneAddress`

Reply:
It's not a boolean but rather the only address if there's exactly one or undefined otherwise

Reply:
Ok, but I still think hint is not the best name. `soleAddress`?

Reply:
Or maybe pass addresses to EmojiInput and check it there

Reply:
Renamed to `soleAddress`